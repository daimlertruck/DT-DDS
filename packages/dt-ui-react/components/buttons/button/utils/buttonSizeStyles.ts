import { CustomTheme as Theme } from '../../../../types';
import { ButtonSizes } from '../constants/types';

export const buttonSizeStyles = (size: ButtonSizes, theme: Theme) =>
  ({
    large: `
      ${theme.fontStyles.button1}
      padding: ${theme.spacing['2xs']} ${theme.spacing.s};
    `,
    medium: `
      ${theme.fontStyles.button2}
      padding: ${theme.spacing['3xs']} ${theme.spacing.xs};
    `,
    small: `
      ${theme.fontStyles.button2}
      padding: ${theme.spacing['4xs']} ${theme.spacing['2xs']};
    `,
  }[size]);
