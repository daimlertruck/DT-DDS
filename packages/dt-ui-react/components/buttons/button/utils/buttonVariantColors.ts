import { CustomTheme as Theme } from '../../../../types';
import {
  ButtonVariant,
  ContainedColors,
  OutlinedColors,
  TextColors,
} from '../constants/types';

interface ColorKeys {
  color: string;
  bgColor: string;
  hoverBgColor?: string;
  hoverColor?: string;
}

export const buttonVariantColors = (
  color: ContainedColors | OutlinedColors,
  theme: Theme
): Record<ButtonVariant, ColorKeys> => ({
  contained: {
    color: theme.colors.alpha.white,
    bgColor: theme.palette.alpha[color],
    hoverBgColor: theme.palette.alpha[`${color}Dark_300` as ContainedColors],
  },
  outlined: {
    color:
      color === 'neutral'
        ? theme.colors.alpha.gray_200
        : theme.palette.alpha[color],
    bgColor: theme.colors.alpha.white,
    hoverBgColor: theme.palette.alpha[`${color}Light_300` as OutlinedColors],
  },
  text: {
    color: theme.palette.alpha[color],
    bgColor: 'transparent',
    hoverColor: theme.palette.alpha[`${color as TextColors}Dark_300`],
  },
});
