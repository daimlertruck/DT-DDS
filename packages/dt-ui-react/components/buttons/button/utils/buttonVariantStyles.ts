import { CustomTheme as Theme } from '../../../../types';
import { ButtonVariant, OutlinedColors, SolidColors } from '../constants/types';

export const buttonVariantStyles = (
  variant: ButtonVariant,
  theme: Theme,
  color: SolidColors | OutlinedColors
) =>
  ({
    solid: `
      text-transform: uppercase;
      border: ${theme.spacing.none};

      &:focus {
        box-shadow: 0px 2px 1px 0px #1A1A1A0D, 0px 0px 0px 4px ${theme.colors.blue_10};
        background-color: ${theme.palette[color]};
      }

      &:disabled {
        opacity: 38%;
      }
  `,
    outlined: `
      border: 1px solid;
      text-transform: uppercase;
        ${
          color === 'neutral'
            ? theme.palette.neutralDark_100
            : theme.palette[color]
        };

      &:disabled {
        color: ${
          color === 'neutral'
            ? theme.palette.neutralLight_100
            : theme.palette[color]
        };
        background-color: ${theme.colors.white};
        border-color: ${
          color === 'neutral'
            ? theme.palette.neutralLight_100
            : theme.palette[color]
        };
        cursor: default;
      }
    `,
    text: `
      border: none;

      &:disabled {
        color: ${theme.palette.neutralLight_100};
      }
    `,
  }[variant]);
