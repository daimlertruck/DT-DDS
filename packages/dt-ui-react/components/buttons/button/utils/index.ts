export * from './buttonVariantColors';
export * from './buttonVariantStyles';
