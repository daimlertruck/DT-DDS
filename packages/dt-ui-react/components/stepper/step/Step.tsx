import { useTheme } from '@emotion/react';
import { Children, cloneElement, ReactElement, useMemo } from 'react';

import { Check } from '../../../core';
import { BaseProps } from '../../../types';
import { Counter } from '../../counter';
import { Typography } from '../../typography';

import { LabelStyled, StepStyled } from './Step.styled';

interface BaseStepProps extends Pick<BaseProps, 'children'> {
  isActive?: boolean;
  isCompleted?: boolean;
  isDisabled?: boolean;
  isError?: boolean;
}

const Step = ({
  children,
  isActive = false,
  isCompleted = false,
  isDisabled = false,
  isError = false,
}: BaseStepProps) => {
  const clonedChildren = useMemo(
    () =>
      Children.map(children as ReactElement<BaseStepProps>, (child) => {
        return (
          child &&
          cloneElement(child, {
            isActive,
            isCompleted,
            isDisabled,
            isError,
            ...child.props,
          })
        );
      }),
    [children, isActive, isCompleted, isDisabled, isError]
  );

  return <StepStyled>{clonedChildren}</StepStyled>;
};

const StepCounter = ({
  children,
  isActive = false,
  isCompleted = false,
  isDisabled = false,
  isError = false,
}: BaseStepProps) => {
  const theme = useTheme();
  const color = isError ? 'error' : isDisabled ? 'neutralLight_100' : 'primary';
  const bgColor = isError
    ? theme.palette.errorLight_300
    : theme.palette.primaryLight_300;

  return (
    <Counter
      color={color}
      isLarge
      outlined={!isCompleted}
      style={{
        borderWidth: 1,
        ...(isActive && { backgroundColor: bgColor }),
      }}
    >
      {isCompleted ? (
        <Check height={8} width={10} />
      ) : (
        <Typography
          color={color}
          element='span'
          fontStyles='pXXSmall'
          style={{ fontWeight: 700 }}
        >
          {children}
        </Typography>
      )}
    </Counter>
  );
};

Step.Counter = StepCounter;

const labelColor = ({
  isActive,
  isError,
  isDisabled,
  isCompleted,
}: Partial<BaseStepProps>) => {
  if (isError) {
    return 'error';
  } else if (isActive || isCompleted) {
    return 'primary';
  } else if (isDisabled) {
    return 'neutralLight_100';
  } else {
    return 'neutral';
  }
};

Step.Label = ({
  children,
  isActive = false,
  isCompleted = false,
  isDisabled = false,
  isError = false,
}: BaseStepProps) => {
  const color = labelColor({ isActive, isError, isDisabled, isCompleted });

  return <LabelStyled color={color}>{children}</LabelStyled>;
};

export default Step;
