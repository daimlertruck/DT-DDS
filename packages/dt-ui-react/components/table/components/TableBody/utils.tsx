import { Children, ReactNode, cloneElement, isValidElement } from 'react';

import { TableDataCellHeader } from '../TableDataCell/TableDataCell';

export const addHeadersToDataCells = (
  dataCells: ReactNode,
  headers: ReactNode[]
) => {
  return Children.map(dataCells, (dataCell, index) => {
    if (isValidElement(dataCell)) {
      const header = headers[index];
      return cloneElement(
        dataCell,
        {},
        <>
          <TableDataCellHeader>{header}</TableDataCellHeader>
          {dataCell.props.children}
        </>
      );
    }
    return dataCell;
  });
};

export const addHeadersToDataCellsInRows = (
  rows: ReactNode,
  headers: ReactNode[]
) => {
  return Children.map(rows, (row) => {
    if (isValidElement(row)) {
      return cloneElement(
        row,
        {},
        addHeadersToDataCells(row.props.children, headers)
      );
    }
    return row;
  });
};
