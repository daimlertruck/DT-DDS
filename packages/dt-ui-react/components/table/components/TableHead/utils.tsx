import { Children, ReactNode, isValidElement } from 'react';

const extractColumnHeaders = (columnHeaders: ReactNode): ReactNode[] => {
  return Children.toArray(columnHeaders).reduce<ReactNode[]>(
    (headers, columnHeader) => {
      if (isValidElement(columnHeader)) {
        return [...headers, columnHeader.props.children];
      }

      return headers;
    },
    []
  );
};

export const extractTableHeaders = (theadRow: ReactNode) => {
  if (isValidElement(theadRow)) {
    return extractColumnHeaders(theadRow.props.children);
  }

  return [];
};
