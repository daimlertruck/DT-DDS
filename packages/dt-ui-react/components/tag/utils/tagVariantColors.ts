import { CustomTheme as Theme } from '../../../types';
import { TagColor, TagVariant } from '../constants';

interface TagVariantColors {
  border: string | number;
  color: string;
  borderColor?: string;
  backgroundColor: string;
  '&:hover'?: {
    borderColor?: string;
    backgroundColor: string;
    color: string;
  };
}

interface TagVariantColorsProps {
  theme: Theme;
  color: TagColor;
  variant: TagVariant;
  hasHover: boolean;
  isDisabled: boolean;
}

const getColorWithShade = ({
  theme,
  color,
  shade,
}: Pick<TagVariantColorsProps, 'color'> & {
  shade: '90' | '60' | '30' | '10';
  theme: Theme;
}) => {
  const { colors, palette } = theme;

  if (color === 'primary') {
    return {
      '90': palette.primaryDark_500,
      '60': palette.primaryDark_200,
      '30': palette.primaryLight_100,
      '10': palette.primaryLight_400,
    }[shade];
  }

  const colorWithShade: keyof Theme['colors'] = `${color}_${shade}`;

  return colors[colorWithShade];
};

export const tagVariantColors = ({
  theme,
  color,
  variant,
  hasHover,
  isDisabled,
}: TagVariantColorsProps): TagVariantColors => {
  const colorShade90 = getColorWithShade({
    theme,
    color,
    shade: '90',
  });
  const colorShade60 = getColorWithShade({
    theme,
    color,
    shade: '60',
  });
  const colorShade30 = getColorWithShade({
    theme,
    color,
    shade: '30',
  });
  const colorShade10 = getColorWithShade({
    theme,
    color,
    shade: '10',
  });

  const colorWithDisabled = isDisabled ? colorShade30 : colorShade60;

  return {
    solid: {
      border: theme.spacing.none,
      color: theme.colors.white,
      backgroundColor: colorWithDisabled,
      ...(hasHover && {
        '&:hover': {
          backgroundColor: colorShade90,
          color: theme.colors.white,
        },
      }),
    },
    outlined: {
      border: `1px solid ${colorShade60}`,
      color: colorWithDisabled,
      backgroundColor: colorShade10,
      borderColor: colorWithDisabled,
      ...(hasHover && {
        '&:hover': {
          borderColor: colorShade90,
          backgroundColor: colorShade10,
          color: colorShade90,
        },
      }),
    },
  }[variant];
};
