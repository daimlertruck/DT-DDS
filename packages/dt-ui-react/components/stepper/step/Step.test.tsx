import { render } from '@testing-library/react';
import React from 'react';

import { withProviders } from '../../../utils';

import Step from './Step';

describe('<Step /> component', () => {
  const ProvidedStep = withProviders(Step);

  describe('render', () => {
    it('default step', () => {
      const { container } = render(
        <ProvidedStep>
          <Step.Counter>1</Step.Counter>
          <Step.Label>Label</Step.Label>
        </ProvidedStep>
      );

      expect(container).toMatchSnapshot();
    });

    it('active step', () => {
      const { container } = render(
        <ProvidedStep isActive>
          <Step.Counter>1</Step.Counter>
          <Step.Label>Label</Step.Label>
        </ProvidedStep>
      );

      expect(container).toMatchSnapshot();
    });

    it('disabled step', () => {
      const { container } = render(
        <ProvidedStep isDisabled>
          <Step.Counter>1</Step.Counter>
          <Step.Label>Label</Step.Label>
        </ProvidedStep>
      );

      expect(container).toMatchSnapshot();
    });

    it('error step', () => {
      const { container } = render(
        <ProvidedStep isError>
          <Step.Counter>1</Step.Counter>
          <Step.Label>Label</Step.Label>
        </ProvidedStep>
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe('<Step.Counter /> component', () => {
    const ProvidedStepCounter = withProviders(Step.Counter);
    it('should render default behaviour', () => {
      const { container } = render(
        <ProvidedStepCounter>1</ProvidedStepCounter>
      );

      expect(container).toMatchSnapshot();
    });

    it('should render active behaviour', () => {
      const { container } = render(
        <ProvidedStepCounter isActive>1</ProvidedStepCounter>
      );

      expect(container).toMatchSnapshot();
    });

    it('should render completed behaviour', () => {
      const { container } = render(
        <ProvidedStepCounter isCompleted>1</ProvidedStepCounter>
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe('<Step.Label /> component', () => {
    const ProvidedStepLabel = withProviders(Step.Label);
    it('should render default behaviour', () => {
      const { container } = render(
        <ProvidedStepLabel>Label</ProvidedStepLabel>
      );

      expect(container).toMatchSnapshot();
    });

    it('should render active behaviour', () => {
      const { container } = render(
        <ProvidedStepLabel isActive>Label</ProvidedStepLabel>
      );

      expect(container).toMatchSnapshot();
    });

    it('should render completed behaviour', () => {
      const { container } = render(
        <ProvidedStepLabel isCompleted>1</ProvidedStepLabel>
      );

      expect(container).toMatchSnapshot();
    });
  });
});
