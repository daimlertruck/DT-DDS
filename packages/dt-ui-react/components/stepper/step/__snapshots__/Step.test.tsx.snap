// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Step /> component <Step.Counter /> component should render active behaviour 1`] = `
.emotion-0 {
  font-weight: 700;
  font-size: 10px;
  line-height: 13.4px;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #FFFFFF;
  border: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 24px;
  height: 24px;
  border-radius: 50%;
  background-color: #036DC1;
  border: 2px solid #036DC1;
  color: #036DC1;
  background-color: #FFFFFF;
}

.emotion-2 {
  font-weight: 400;
  font-size: 12px;
  line-height: 16.08px;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #036DC1;
}

<div>
  <span
    class="emotion-0 emotion-1"
    color="primary"
    data-testid="counter"
    style="border-width: 1px; background-color: rgb(204, 232, 255);"
  >
    <span
      class="emotion-2 emotion-3"
      color="primary"
      data-testid="typography"
      style="font-weight: 700;"
    >
      1
    </span>
  </span>
</div>
`;

exports[`<Step /> component <Step.Counter /> component should render completed behaviour 1`] = `
.emotion-0 {
  font-weight: 700;
  font-size: 10px;
  line-height: 13.4px;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #FFFFFF;
  border: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 24px;
  height: 24px;
  border-radius: 50%;
  background-color: #036DC1;
}

<div>
  <span
    class="emotion-0 emotion-1"
    color="primary"
    data-testid="counter"
    style="border-width: 1px;"
  >
    <svg
      fill="none"
      height="8"
      viewBox="0 0 10 8"
      width="10"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 4.818 3.857 7 9 1"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
  </span>
</div>
`;

exports[`<Step /> component <Step.Counter /> component should render default behaviour 1`] = `
.emotion-0 {
  font-weight: 700;
  font-size: 10px;
  line-height: 13.4px;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #FFFFFF;
  border: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 24px;
  height: 24px;
  border-radius: 50%;
  background-color: #036DC1;
  border: 2px solid #036DC1;
  color: #036DC1;
  background-color: #FFFFFF;
}

.emotion-2 {
  font-weight: 400;
  font-size: 12px;
  line-height: 16.08px;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #036DC1;
}

<div>
  <span
    class="emotion-0 emotion-1"
    color="primary"
    data-testid="counter"
    style="border-width: 1px;"
  >
    <span
      class="emotion-2 emotion-3"
      color="primary"
      data-testid="typography"
      style="font-weight: 700;"
    >
      1
    </span>
  </span>
</div>
`;

exports[`<Step /> component <Step.Label /> component should render active behaviour 1`] = `
.emotion-0 {
  font-weight: 400;
  font-size: 16px;
  line-height: 21.44px;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #036DC1;
}

<div>
  <span
    class="emotion-0 emotion-1"
    color="primary"
  >
    Label
  </span>
</div>
`;

exports[`<Step /> component <Step.Label /> component should render completed behaviour 1`] = `
.emotion-0 {
  font-weight: 400;
  font-size: 16px;
  line-height: 21.44px;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #036DC1;
}

<div>
  <span
    class="emotion-0 emotion-1"
    color="primary"
  >
    1
  </span>
</div>
`;

exports[`<Step /> component <Step.Label /> component should render default behaviour 1`] = `
.emotion-0 {
  font-weight: 400;
  font-size: 16px;
  line-height: 21.44px;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #EBECED;
}

<div>
  <span
    class="emotion-0 emotion-1"
    color="neutral"
  >
    Label
  </span>
</div>
`;

exports[`<Step /> component render active step 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 12px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  font-weight: 700;
  font-size: 10px;
  line-height: 13.4px;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #FFFFFF;
  border: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 24px;
  height: 24px;
  border-radius: 50%;
  background-color: #036DC1;
  border: 2px solid #036DC1;
  color: #036DC1;
  background-color: #FFFFFF;
}

.emotion-4 {
  font-weight: 400;
  font-size: 12px;
  line-height: 16.08px;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #036DC1;
}

.emotion-6 {
  font-weight: 400;
  font-size: 16px;
  line-height: 21.44px;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #036DC1;
}

<div>
  <li
    class="emotion-0 emotion-1"
  >
    <span
      class="emotion-2 emotion-3"
      color="primary"
      data-testid="counter"
      style="border-width: 1px; background-color: rgb(204, 232, 255);"
    >
      <span
        class="emotion-4 emotion-5"
        color="primary"
        data-testid="typography"
        style="font-weight: 700;"
      >
        1
      </span>
    </span>
    <span
      class="emotion-6 emotion-7"
      color="primary"
    >
      Label
    </span>
  </li>
</div>
`;

exports[`<Step /> component render default step 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 12px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  font-weight: 700;
  font-size: 10px;
  line-height: 13.4px;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #FFFFFF;
  border: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 24px;
  height: 24px;
  border-radius: 50%;
  background-color: #036DC1;
  border: 2px solid #036DC1;
  color: #036DC1;
  background-color: #FFFFFF;
}

.emotion-4 {
  font-weight: 400;
  font-size: 12px;
  line-height: 16.08px;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #036DC1;
}

.emotion-6 {
  font-weight: 400;
  font-size: 16px;
  line-height: 21.44px;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #EBECED;
}

<div>
  <li
    class="emotion-0 emotion-1"
  >
    <span
      class="emotion-2 emotion-3"
      color="primary"
      data-testid="counter"
      style="border-width: 1px;"
    >
      <span
        class="emotion-4 emotion-5"
        color="primary"
        data-testid="typography"
        style="font-weight: 700;"
      >
        1
      </span>
    </span>
    <span
      class="emotion-6 emotion-7"
      color="neutral"
    >
      Label
    </span>
  </li>
</div>
`;

exports[`<Step /> component render disabled step 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 12px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  font-weight: 700;
  font-size: 10px;
  line-height: 13.4px;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #FFFFFF;
  border: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 24px;
  height: 24px;
  border-radius: 50%;
  background-color: #F8F9FA;
  border: 2px solid #F8F9FA;
  color: #F8F9FA;
  background-color: #FFFFFF;
}

.emotion-4 {
  font-weight: 400;
  font-size: 12px;
  line-height: 16.08px;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #F8F9FA;
}

.emotion-6 {
  font-weight: 400;
  font-size: 16px;
  line-height: 21.44px;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #F8F9FA;
}

<div>
  <li
    class="emotion-0 emotion-1"
  >
    <span
      class="emotion-2 emotion-3"
      color="neutralLight_100"
      data-testid="counter"
      style="border-width: 1px;"
    >
      <span
        class="emotion-4 emotion-5"
        color="neutralLight_100"
        data-testid="typography"
        style="font-weight: 700;"
      >
        1
      </span>
    </span>
    <span
      class="emotion-6 emotion-7"
      color="neutralLight_100"
    >
      Label
    </span>
  </li>
</div>
`;

exports[`<Step /> component render error step 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 12px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  font-weight: 700;
  font-size: 10px;
  line-height: 13.4px;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #FFFFFF;
  border: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 24px;
  height: 24px;
  border-radius: 50%;
  background-color: #FF4A4A;
  border: 2px solid #FF4A4A;
  color: #FF4A4A;
  background-color: #FFFFFF;
}

.emotion-4 {
  font-weight: 400;
  font-size: 12px;
  line-height: 16.08px;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #FF4A4A;
}

.emotion-6 {
  font-weight: 400;
  font-size: 16px;
  line-height: 21.44px;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #FF4A4A;
}

<div>
  <li
    class="emotion-0 emotion-1"
  >
    <span
      class="emotion-2 emotion-3"
      color="error"
      data-testid="counter"
      style="border-width: 1px;"
    >
      <span
        class="emotion-4 emotion-5"
        color="error"
        data-testid="typography"
        style="font-weight: 700;"
      >
        1
      </span>
    </span>
    <span
      class="emotion-6 emotion-7"
      color="error"
    >
      Label
    </span>
  </li>
</div>
`;
